compiler/fir/tree/src/org/jetbrains/kotlin/fir/expressions/ExhaustivenessStatus.kt

```diff
@@ -8,7 +8,19 @@ package org.jetbrains.kotlin.fir.expressions
 import org.jetbrains.kotlin.diagnostics.WhenMissingCase
 
 sealed class ExhaustivenessStatus {
-    object Exhaustive : ExhaustivenessStatus()
+
+    /**
+     * This value is used if the subject has type other than `Nothing`, in which case it's literally exhaustive only if type's possible
+     * cases are properly covered.
+     */
+    object ProperlyExhaustive : ExhaustivenessStatus()
+
+    /**
+     *  This value is used if the subject has type `Nothing`, in which case even an empty `when` is considered exhaustive. Also, in this
+     *  case, a synthetic else branch is created.
+     */
+    object TriviallyExhaustive : ExhaustivenessStatus()
```

Comment:
May be ExhaustiveAsNothing is a better name?

Reply:
Sure, sounds good!